src/vid/advz.rs

```diff
@@ -217,11 +223,24 @@ where
         let aggregate_eval =
             polynomial_eval(share.evals.iter().map(FieldMultiplier), pseudorandom_scalar);
 
+        // prepare eval point for aggregate proof
+        let domain =
+            P::multi_open_rou_eval_domain(self.payload_chunk_size, self.num_storage_nodes)?;
+        // TODO(Gus) perf: `nth` scales linearly with number of storage nodes!
+        // https://github.com/EspressoSystems/jellyfish/issues/313
+        let point = domain.elements().nth(share.index).ok_or_else(|| {
```

Comment:
Why not directly using `domain.nth()` here?

Reply:
I don't see a `nth` method. But looking more closely I do see a `element(i)` method that I did not notice earlier. I'll use that.

Reply:
done in https://github.com/EspressoSystems/hotshot-primitives/pull/56/commits/970384dafb57f417e3d5c30e16653c10afaf9c56